docs/en/observability/cloud-monitoring/aws/monitor-aws-firewall-firehose.asciidoc

```diff
@@ -0,0 +1,120 @@
+[[monitor-aws-firewall-firehose]]
+= Monitor AWS Network Firewall logs
+
+++++
+<titleabbrev>Monitor Network Firewall logs</titleabbrev>
+++++
+
+In this section, you'll learn how to send AWS Network Firewall log events from AWS to your Elastic stack using Amazon Data Firehose.
+
+You will go through the following steps:
+
+- Select a AWS Network Firewall-compatible resource 
+- Create a delivery stream in Amazon Data Firehose
+- Set up logging to forward the logs to the Elastic stack using a Firehose stream
+- Visualize your logs in {kib}
+
+[discrete]
+[[firehose-firewall-prerequisites]]
+== Before you begin
+
+We assume that you already have:
+
+- An AWS account with permissions to pull the necessary data from AWS.
+- A deployment using our hosted {ess} on {ess-trial}[{ecloud}]. The deployment includes an {es} cluster for storing and searching your data, and {kib} for visualizing and managing your data. AWS Data Firehose works with Elastic Stack version 7.17 or greater, running on Elastic Cloud only.
+
+IMPORTANT: AWS PrivateLink is not supported. Make sure the deployment is on AWS, because the Amazon Data Firehose delivery stream connects specifically to an endpoint that needs to be on AWS.
+
+[discrete]
+[[firehose-firewall-step-one]]
+== Step 1: Install AWS integration in {kib}
+
+. In {kib}, navigate to *Management* > *Integrations* and browse the catalog to find the AWS integration.
+
+. Navigate to the *Settings* tab and click *Install AWS assets*.
+
+[discrete]
+[[firehose-firewall-step-two]]
+== Step 2: Select a resource
+
+[role="screenshot"]
+image::firehose-networkfirewall-firewall.png[AWS Network Firewall]
+
+You can either use an existing AWS Network Firewall, or create a new one for testing purposes.
+
+Creating a Network Firewall is not trivial and is beyond the scope of this guide. For more information, see the AWS documentation on the https://docs.aws.amazon.com/network-firewall/latest/developerguide/getting-started.html[Getting started with AWS Network Firewall] guide.
+
```

Comment:
@dedemorton, explaining how to set up a Network Firewall for testing would be 2x the size of this guide.

So, I guess we have (at least) two options:

- say, "it's beyond the scope of this guide."
- link to an external source 

I created a Terraform module at https://github.com/zmoog/integrations-cookbook/pull/1/files to test the Network Firewall for this guide. The module would allow users to set up the entire Network Firewall in one step. 

However, (1) the module is in draft on (2) my personal repo. For the draft, I can finish it up quickly. But I can't link to my personal repo from the official Elastic docs.

What are our options here? Can we deliver the Terraform module through an official Elastic resource? Any suggestion is welcome!

Reply:
Good question! TBH it's been quite awhile since I've provided users with links to special resources. Does it need to be a GitHub repo? I know in the past we've used download.elastic.co to provide sample data for guides, but this seems different. @bmorelli25 Do you have any ideas for the best way to deliver the Terraform module that Maurizio created for testing the network firewall?

Reply:
Hi @zmoog. Yes, we have the https://github.com/elastic/observability-examples repo for exactly this purpose.